noodles-editor/src/components/settings-dialog.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState, useEffect } from 'react';
+import * as Dialog from '@radix-ui/react-dialog';
+import { Cross2Icon } from '@radix-ui/react-icons';
+import { analytics } from '../utils/analytics';
+import s from './settings-dialog.module.css';
+
+interface SettingsDialogProps {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+}
+
+export function SettingsDialog({ open, setOpen }: SettingsDialogProps) {
+  const [analyticsEnabled, setAnalyticsEnabled] = useState(false);
+
+  useEffect(() => {
+    const consent = analytics.getConsent();
+    setAnalyticsEnabled(consent?.enabled ?? false);
+  }, [open]);
+
+  const handleAnalyticsToggle = (enabled: boolean) => {
+    setAnalyticsEnabled(enabled);
+    analytics.setConsent(enabled);
+
+    if (enabled) {
+      analytics.track('analytics_enabled_in_settings');
+    }
+  };
+
+  return (
+    <Dialog.Root open={open} onOpenChange={setOpen}>
+      <Dialog.Portal>
+        <Dialog.Overlay className={s.overlay} />
+        <Dialog.Content className={s.content}>
+          <Dialog.Title className={s.title}>Settings</Dialog.Title>
+
+          <div className={s.section}>
+            <h3 className={s.sectionTitle}>Privacy & Analytics</h3>
+
+            <div className={s.settingItem}>
+              <label className={s.settingLabel}>
+                <input
+                  type="checkbox"
+                  checked={analyticsEnabled}
+                  onChange={(e) => handleAnalyticsToggle(e.target.checked)}
+                  className={s.checkbox}
+                />
+                <div className={s.settingContent}>
+                  <div className={s.settingName}>Share anonymous usage data</div>
+                  <div className={s.settingDescription}>
+                    Help improve Noodles.gl by sharing anonymous feature usage data. We never
+                    collect your project data, node content, API keys, or personal information.
```

Comment:
Might be a good idea to link to our privacy policy here https://www.jobyaviation.com/privacy-policy/

Reply:
Maybe we add a PRIVACY.md file?